File: leetcode/3100-water-bottles-ii/solution.cpp
/**
 * Author: 1chooo<hugo@1chooo.com>
 * Problem: https://leetcode.com/problems/water-bottles
 * Runtime: 0ms (100.00%)
 */

class Solution {
  public:
    int maxBottlesDrunk(int numBottles, int numExchange) {
        int total_bottle = numBottles;

        while (numBottles >= numExchange) {
            numBottles -= numExchange;
            total_bottle++;
            numBottles++;
            numExchange++;
        }

        return total_bottle;
    }
};

File: leetcode/3100-water-bottles-ii/solution.py
"""
* Author: 1chooo<hugo@1chooo.com>
* Problem: https://leetcode.com/problems/water-bottles
* Runtime: 26ms (98.62%)
"""


class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        total_drunk = 0
        empty = 0

        while numBottles > 0:
            total_drunk += numBottles
            empty += numBottles

            numBottles = 0

            if empty >= numExchange:
                empty -= numExchange
                numBottles += 1
                numExchange += 1
            else:
                break

        return total_drunk
